subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorParallelIntegrationTest.groovy

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal
+
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.junit.Rule
+import spock.lang.Timeout
+import spock.lang.Unroll
+
+@Timeout(60)
+class WorkerExecutorParallelIntegrationTest extends AbstractWorkerExecutorIntegrationTest {
```

Comment:
good test coverage.

We should add tests that show:

- A task that depends on a task that runs a bunch of actions does not start until the actions have been completed.
- `--max-workers` is honored when running actions in parallel.
- That a TAPI client receives build operation events for each of the task actions and the work actions.